app/controllers/api/messaging.php

```diff
@@ -2945,7 +2945,11 @@
                 throw new Exception(Exception::STORAGE_FILE_NOT_PUBLIC);
             }
 
-            $image = "{$protocol}://{$host}/v1/storage/buckets/{$bucket->getId()}/files/{$file->getId()}/view?project={$project->getId()}";
+            $image = [
+                'bucketId' => $bucket->getId(),
+                'fileId' => $file->getId(),
+                'url' => "{$protocol}://{$host}/v1/storage/buckets/{$bucket->getId()}/files/{$file->getId()}/view?project={$project->getId()}",
+            ];
```

Comment:
why did we need to do this?

Reply:
the data is destroyed for the frontend to properly display it, if the compoundId gets saved as the preview URL - I cannot update it again with a compundId 👍🏻